Tools/llm-tool/LLMTool.swift

```diff
@@ -85,16 +90,14 @@ struct GenerateArguments: ParsableArguments, Sendable {
     }
 
     func generate(
-        promptTokens: [Int], model: LLMModel, tokenizer: Tokenizer,
-        extraEOSTokens: Set<String>? = nil
+        input: LMInput, context: ModelContext
     )
-        -> GenerateResult
+        throws -> GenerateResult
     {
-        var detokenizer = NaiveStreamingDetokenizer(tokenizer: tokenizer)
+        var detokenizer = NaiveStreamingDetokenizer(tokenizer: context.tokenizer)
 
-        return LLM.generate(
-            promptTokens: promptTokens, parameters: generateParameters,
-            model: model, tokenizer: tokenizer, extraEOSTokens: extraEOSTokens
+        return try MLXLMCommon.generate(
+            input: input, parameters: generateParameters, context: context
```

Comment:
This is just moving to the newer API to fix deprecation warnings.  The older code still ran